### PR TITLE
Fix templateUtility not work in required common.js

### PR DESCRIPTION
**Pull request by qinezh**
@chenkennt @superyyrrzz @hellosnow @ansyral

### CODE REVIEW

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/TemplatePreprocessors/TemplateJintPreprocessor.cs [hunk 1]
```diff
@@ -54,6 +54,7 @@ public class TemplateJintPreprocessor : ITemplatePreprocessor
         private const string RequireFuncVariableName = "require";
         private const string RequireRelativePathPrefix = "./";
 
+        private static object _utilityObject;
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/TemplatePreprocessors/TemplateJintPreprocessor.cs]: why static?

File: src/Microsoft.DocAsCode.Build.Engine/TemplatePreprocessors/TemplateJintPreprocessor.cs [hunk 2]
```diff
@@ -86,14 +87,16 @@ private Engine SetupEngine(ResourceCollection resourceCollection, TemplatePrepro
             var rootPath = (RelativePath)scriptResource.ResourceName;
             var engineCache = new Dictionary<string, Engine>();
 
-            var utility = new TemplateUtility(context);
-            object utilityObject = new
+            if (_utilityObject == null)
             {
-                resolveSourceRelativePath = new Func<string, string, string>((s1, s2) => utility.ResolveSourceRelativePath(s1, s2))
-            };
+                var utility = new TemplateUtility(context);
```
Comment (chenkennt) [inline src/Microsoft.DocAsCode.Build.Engine/TemplatePreprocessors/TemplateJintPreprocessor.cs]: >context [](start = 50, length = 7)

`_utilityObject` is static but `context` is not. Will context change between different calls of this method?

**Comment by ansyral** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/TemplatePreprocessors/TemplateJintPreprocessor.cs [hunk 3]
```diff
@@ -87,13 +88,12 @@ private Engine SetupEngine(ResourceCollection resourceCollection, TemplatePrepro
             var engineCache = new Dictionary<string, Engine>();
 
             var utility = new TemplateUtility(context);
-            object utilityObject = new
+            _utilityObject = new
             {
                 resolveSourceRelativePath = new Func<string, string, string>((s1, s2) => utility.ResolveSourceRelativePath(s1, s2))
```
Comment (chenkennt) [inline src/Microsoft.DocAsCode.Build.Engine/TemplatePreprocessors/TemplateJintPreprocessor.cs]: >(s1, s2) => utility.ResolveSourceRelativePath(s1, s2) [](start = 77, length = 53)

just `utility.ResolveSourceRelativePath`?

**Comment by chenkennt** [comment]
:shipit:

**Pull request merged**